### ISSUE TITLE
 items.scaleSelect.options is not iterable error

### ISSUE THREAD

**Issue by tarun04** [stephanrauh/ngx-extended-pdf-viewer#257]
Code:
![image](https://user-images.githubusercontent.com/14960107/78711494-0da54400-78e5-11ea-80e0-f76814cb8bf6.png)

Error:
![image](https://user-images.githubusercontent.com/14960107/78710750-e00bcb00-78e3-11ea-8ba0-fb91fc3fa557.png)

**Comment by stephanrauh** [comment]
I suppose you're talking about these lines:

```javascript
    const customScale = Math.round(pageScale * 10000) / 100;
    this.l10n.get("page_scale_percent", {
      scale: customScale
    }, "{{scale}}%").then(msg => {
      let predefinedValueFound = false;

      for (const option of items.scaleSelect.options) {
        if (option.value !== pageScaleValue) {
          option.selected = false;
          continue;
        }

        option.selected = true;
        predefinedValueFound = true;
      }
```
I can't imagine how this `for` loop can break. The only idea I have is you've removed the dropdown from the toolbar. But I don't see any customization in your sourcecode. Weird.

On pdfviewer.net, this compiles to something completely different. That version looks even more resilient:

```javascript
    this.l10n.get('page_scale_percent', {
      scale: customScale
    }, '{{scale}}%').then(msg => {
      let options = items.scaleSelect.options;
      let predefinedValueFound = false;

      for (let i = 0, ii = options.length; i < ii; i++) {
        let option = options[i];

        if (option.value !== pageScaleValue) {
          option.selected = false;
          continue;
        }

        option.selected = true;
        predefinedValueFound = true;
      }
```

**Comment by stephanrauh** [comment]
Please add a breakpoint to line 11670 of `viewer.js`.  Which value does `items.scaleSelect.options` have? Is it undefined? If not, which class is it? In other words, what is `items.scaleSelect.options.constructor.name`?

**Comment by tarun04** [comment]
> Please add a breakpoint to line 11670 of `viewer.js`. Which value does `items.scaleSelect.options` have? Is it undefined? If not, which class is it? In other words, what is `items.scaleSelect.options.constructor.name`?

It is undefined

**Comment by tarun04** [comment]
I have a pdfViewerComponent which is a dialog.
When I click a button, it calls API gets the base64 (pdfContent) and opens the dialog where 
` <ngx-extended-pdf-viewer>... </....> ` is loaded after the pdfContent is set to have some value.

![image](https://user-images.githubusercontent.com/14960107/78719105-61b62580-78f1-11ea-9309-65d6fca8a3b6.png)

I can't really create a reproducer atm.

**Comment by stephanrauh** [comment]
Hm. I expected `undefined`. But I don't know how to solve the problem. Or how to reproduce it.

Maybe it helps to add the attribute `[delayFirstView]="1000"`. That solves many timing problems and race conditions. It's a bad solution, but please give it a try. If it works, that might be a clue how to find and solve the root cause.

**Comment by stephanrauh** [comment]
BTW, I'm 90% sure this bug doesn't have to do with Base64 files. Would you mind to run a test with a static PDF in the `assets` folder?

**Comment by tarun04** [comment]
I will try both suggestions and get back to you. 
Thank you for immediate response.

**Comment by tarun04** [comment]
> Hm. I expected `undefined`. But I don't know how to solve the problem. Or how to reproduce it.
> 
> Maybe it helps to add the attribute `[delayFirstView]="1000"`. That solves many timing problems and race conditions. It's a bad solution, but please give it a try. If it works, that might be a clue how to find and solve the root cause.

delayFirstView didn't work.

But when I removed [customToolbar] it worked fine.

**Comment by tarun04** [comment]
So the issue is with the ng-template. It has to have one of the components that are in the setup provided.

![image](https://user-images.githubusercontent.com/14960107/78794596-f700fa80-7981-11ea-98bd-ce4298c6eff1.png)

But I want a custom view for it. So as a workaround, I just included <pdf-zoom-toolbar ></pdf-zoom-toolbar> with display set to none in css. It worked.

**Comment by stephanrauh** [comment]
Yeah, that's the work-around I wanted to suggest you, too.

Sorry for missing the `[customToolbar]` bit. It was a long working day yesterday, so I was very tired. I remember looking for something like `[customToolbar]`, but I missed it nonetheless.

Now that I've understood the root cause, I can fix it. Basically, I'm going to insert an invisible and empty zoom dropdown. That approach makes the underlying library, pdf.js, happy because it finds the `id`s it relies on. At the same time, it gives you the freedom to omit the dropdown if you don't need it.

I'm putting this high on my priority list - so your feature is going to land in a few days (if everything goes according to the plan).

Best regards,
Stephan

**Comment by stephanrauh** [comment]
The bugfix (or feature) has landed with version 3.2.2.
